add universal tasks

inveker opened the pull request:
Hi, I've added a tracer for the "deploy" task from the hardhat-deploy plugin.
Is it possible to consider a universal solution so that new tasks do not have to copy-paste the same file?

fvictorio commented:
Keep in mind that this will only work if `hardhat-tracer` is required in the config _after_ `hardhat-deploy`. If it's included before, I think you'll get an error when `runSuper` is called.

> Is it possible to consider a universal solution so that new tasks do not have to copy-paste the same file?

This is a good question. Maybe `wrapHardhatProvider` could be used in `extendEnvironment`, but I'm not really sure. `applyCliArgsToTracer` seems to emulate global args, which Hardhat doesn't support yet.

inveker commented:
Hello, I added the ability to specify the necessary commands in the user's project config. You need to create a tracer.config.json file, at the level of the hardhat.config.json file, and write something like this in it
```json
{
  "tasks": ["deploy"]
}
```

At the same time, I removed code duplicates from task files for "test", "node", "deploy"

zemse commented:
I just add this to the main (v2) branch in [ae4551c](https://github.com/zemse/hardhat-tracer/commit/ae4551c362c08a1fb02ce0afc28ab6e420c1c16d). I think it's better to use the hardhat config instead of having a separate config file for this. Published it temporarily to `hardhat-tracer@2.1.0-beta`.

```
// hardhat.config.ts
{
  tracer: {
    tasks: ['deploy']
  }
}

```
